Demo/src/main/java/com/braintreepayments/demo/BaseFragment.java

```diff
@@ -47,11 +48,15 @@ protected void showDialog(String message) {
         }
     }
 
-    protected BraintreeClient getBraintreeClient() {
+
+    protected void fetchAuthorization(BraintreeAuthorizationCallback callback) {
         DemoActivity demoActivity = getDemoActivity();
         if (demoActivity != null) {
-            return demoActivity.getBraintreeClient();
+            demoActivity.fetchAuthorization(callback);
         }
-        return null;
+    }
```

Comment:
2cents: I always felt like phasing out `BaseFragment` would be the best call. We should be able to fetch client tokens within each feature fragment. Do you feel like it makes sense to remove eventually?

Reply:
Agreed - will look into that in another demo refactor PR